src/main/java/seedu/command/AddCommand.java

```diff
@@ -0,0 +1,32 @@
+package seedu.command;
+
+import com.google.gson.Gson;
+import seedu.module.Module;
+import seedu.timetable.Timetable;
+
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class AddCommand extends Command{
+    private final String ModuleCode;
+    public Timetable timetable = new Timetable();
+
+    public AddCommand(String ModuleCode) {
+        this.ModuleCode = ModuleCode;
+    }
+
+    public void execute() {
+        String path = "data/Modules/" + ModuleCode + ".json";
+        try {
+            Gson gson = new Gson();
+            Reader reader = Files.newBufferedReader(Paths.get(path));
+            Module module = gson.fromJson(reader, Module.class);
+            reader.close();
+            timetable.add(module);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+        timetable.print();
+    }
```

Comment:
Perhaps you could try calling a common fetchMod(String moduleCode) here instead since it seems this might be a frequently used function

Reply:
Amen to that! Yeap I was thinking of implementing that but I just didn't know how it will affect the rest of our team's implementation